File: clients/simpleAgent.js
const WebSocket = require("ws");

// Configuration
const CONFIG = {
  defaultId: "k",
  validDirections: ["up", "down", "left", "right"],
  wsUrl: "ws://localhost:3000"
};

// Get the agentId from command line or use default
const agentId = process.argv[2] || CONFIG.defaultId;

// Connect to WebSocket server
const ws = new WebSocket(`${CONFIG.wsUrl}?id=${agentId}`);

ws.on("open", () => console.log("Connected to WebSocket server"));
ws.on("error", (error) => console.error("WebSocket error:", error));
ws.on("close", () => console.log("Disconnected from WebSocket server"));

ws.on("message", (data) => {
  const gameState = JSON.parse(data.toString("utf-8"));
  console.log("Received game state:", gameState);

  // Only make a move if the game isn't over
  if (gameState.winner === null || gameState.winner === undefined) {
    // Make a random move
    const move = {
      playerId: agentId,
      direction: CONFIG.validDirections[Math.floor(Math.random() * CONFIG.validDirections.length)]
    };

    // Send the move
    ws.send(JSON.stringify(move));
    console.log("Sent move:", move);
  }
});

File: clients/simpleAgent.py
import asyncio
import websockets
import json
import sys
import random

# Configuration
DEFAULT_AGENT_ID = "l"
VALID_DIRECTIONS = ["up", "down", "left", "right"]

async def connect_to_game_server(agent_id):
    uri = f"ws://localhost:3000?id={agent_id}"

    async with websockets.connect(uri) as websocket:
        print("Connected to WebSocket server")
        
        # Receive initial response
        response = await websocket.recv()
        print(f"Initial server response: {response}")

        try:
            while True:
                # Receive game state
                message = await websocket.recv()
                game_state = json.loads(message)
                print(f"Received game state: {game_state}")

                # Check if game is over
                if game_state.get('winner') is not None:
                    print("Game over!")
                    break

                # Make a random move
                move = {
                    "playerId": agent_id,
                    "direction": random.choice(VALID_DIRECTIONS)
                }

                # Send move
                await websocket.send(json.dumps(move))
                print(f"Sent move: {move}")

        except websockets.exceptions.ConnectionClosed:
            print("Disconnected from WebSocket server")
        except Exception as e:
            print(f"Error: {e}")

def main():
    # Get agent ID from command line or use default
    agent_id = sys.argv[1] if len(sys.argv) > 1 else DEFAULT_AGENT_ID
    print(f"Starting agent with ID: {agent_id}")
    
    # Run the client
    asyncio.run(connect_to_game_server(agent_id))

if __name__ == "__main__":
    main()